Source/JavaScriptCore/bytecompiler/BytecodeGenerator.cpp

```diff
@@ -1045,11 +1047,11 @@ BytecodeGenerator::BytecodeGenerator(VM& vm, ModuleProgramNode* moduleProgramNod
     }
 
     emitEnter();
-
     allocateScope();
-    RegisterID* moduleScope = addVar();
-    move(moduleScope, scopeRegister());
```

Comment:
Why previously we didn't do `moduleScope->ref()` but the new code does `m_topLevelScopeRegister->ref()`?

Reply:
Actually it is not necessary, but just following to what scopeRegister is doing. If it is addVar's result, basically it is not necessary.